src/main/kotlin/com/fasterxml/jackson/module/kotlin/KotlinNamesAnnotationIntrospector.kt

```diff
@@ -18,38 +18,42 @@ import java.util.Locale
 import kotlin.reflect.KClass
 import kotlin.reflect.KFunction
 import kotlin.reflect.KParameter
-import kotlin.reflect.full.companionObject
-import kotlin.reflect.full.declaredFunctions
-import kotlin.reflect.full.memberProperties
-import kotlin.reflect.full.primaryConstructor
+import kotlin.reflect.full.*
 import kotlin.reflect.jvm.internal.KotlinReflectionInternalError
+import kotlin.reflect.jvm.javaGetter
 import kotlin.reflect.jvm.javaType
 import kotlin.reflect.jvm.kotlinFunction
 
 internal class KotlinNamesAnnotationIntrospector(val module: KotlinModule, val cache: ReflectionCache, val ignoredClassesForImplyingJsonCreator: Set<KClass<*>>) : NopAnnotationIntrospector() {
     // since 2.4
-    override fun findImplicitPropertyName(member: AnnotatedMember): String? {
-        if (member is AnnotatedMethod && member.isInlineClass()) {
-            if (member.name.startsWith("get") &&
-                    member.name.contains('-') &&
-                    member.parameterCount == 0) {
-                return member.name.substringAfter("get")
-                    .replaceFirstChar { it.lowercase(Locale.getDefault()) }
-                    .substringBefore('-')
-            } else if (member.name.startsWith("is") &&
-                    member.name.contains('-') &&
-                    member.parameterCount == 0) {
-                return member.name.substringAfter("is")
-                    .replaceFirstChar { it.lowercase(Locale.getDefault()) }
-                    .substringBefore('-')
-            }
-        } else if (member is AnnotatedParameter) {
-            return findKotlinParameterName(member)
-        }
-
-        return null
+    override fun findImplicitPropertyName(member: AnnotatedMember): String? = when (member) {
+        is AnnotatedMethod -> findImplicitPropertyNameFromKotlinPropertyIfNeeded(member)
+        is AnnotatedParameter -> findKotlinParameterName(member)
+        else -> null
     }
 
+    // Since getter for value class (inline class) will be compiled into a different name such as "getFoo-${random}",
+    // use the property name in Kotlin in such a case.
+    private fun findImplicitPropertyNameFromKotlinPropertyIfNeeded(member: AnnotatedMethod): String? = member
+        .takeIf {
+            it.parameterCount == 0 &&
+                    it.name.run { contains("-") && (startsWith("get") || startsWith("is")) }
```

Comment:
I think this would be a bit clearer as a method:

```kotlin
fun looksLikeKotlinGeneratedMethod(methodName: String) =
    methodName.run { contains("-") && (startsWith("get") || startsWith("is")) }
```

Reply:
I took it in.
b67bb9a